### ISSUE TITLE
Beta version cannot open existing database using passcode

### ISSUE THREAD

**Issue by Snaptags** [moneymanagerex/android-money-manager-ex#1279]
- use app version v2018.02.26.981 to set up a new database
- open settings/security/acivate passcode
- update app to latest beta
- it will keep crashing until the database has been deleted from the device

I can reproduce this in android studio using the attached data.mmb (passcode: 1234)

[data.zip](https://github.com/moneymanagerex/android-money-manager-ex/files/2752920/data.zip)

Without passcode it opens, after activating the passcode:
![fatal](https://user-images.githubusercontent.com/1249745/51083245-01006300-1717-11e9-93b3-577bb28ce461.png)
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.money.manager.ex.debug, PID: 3376
    java.lang.NoClassDefFoundError: Failed resolution of: Landroid/hardware/fingerprint/FingerprintManager;
        at com.money.manager.ex.PasscodeActivity.onCreate(PasscodeActivity.java:164)
        at android.app.Activity.performCreate(Activity.java:5990)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
        at android.app.ActivityThread.access$800(ActivityThread.java:151)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:135)
        at android.app.ActivityThread.main(ActivityThread.java:5254)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "android.hardware.fingerprint.FingerprintManager" on path: DexPathList[[zip file "/data/app/com.money.manager.ex.debug-2/base.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_dependencies_apk.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_slice_0_apk.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_slice_1_apk.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_slice_2_apk.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_slice_3_apk.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_slice_4_apk.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_slice_5_apk.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_slice_6_apk.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_slice_7_apk.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_slice_8_apk.apk", zip file "/data/app/com.money.manager.ex.debug-2/split_lib_slice_9_apk.apk"],nativeLibraryDirectories=[/vendor/lib, /system/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
        at com.money.manager.ex.PasscodeActivity.onCreate(PasscodeActivity.java:164) 
        at android.app.Activity.performCreate(Activity.java:5990) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387) 
        at android.app.ActivityThread.access$800(ActivityThread.java:151) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:135) 
        at android.app.ActivityThread.main(ActivityThread.java:5254) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:372) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698) 
    	Suppressed: java.lang.ClassNotFoundException: android.hardware.fingerprint.FingerprintManager
        at java.lang.Class.classForName(Native Method)
        at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
        at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
        		... 15 more
     Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack available
E/SQLiteLog: (284) automatic index on sqlite_sq_A3807FB0(id)
E/SQLiteLog: (284) automatic index on sqlite_sq_A3807E70(id)
E/SQLiteLog: (284) automatic index on sqlite_sq_A5C08900(accountid)
```

**Comment by MisterY** [comment]
Oh, that is interesting. Missing fingerprint manager. Probably a check for existence or namespace should be added.

**Comment by velmuruganc** [comment]
Could you please tell me your android version? I will try to recreate the issue at me end. If so, I will try fixing it.

**Comment by Snaptags** [comment]
Android version 8.0.0 (OPSS27.76-12-25-13) running on a Motorola z2 play.

**Comment by MisterY** [comment]
FingerprintManager has been deprecated.

**Comment by wolfsolver** [comment]
Solved in latest release 2024.01.05
